packages/gestalt/src/AbstractEvent.js

```diff
@@ -0,0 +1,14 @@
+// @flow strict
+
+/**
+ * We use a common event payload across all components. Since it's different from standard events that people are used to from React, it's abstracted here to at least make it standard. If you have to learn something, you should only have to learn it once.
+ *
+ * Why is it non-standard? Gestalt components are controlled - meaning that the can contain state and when that changes it's reflected back to the parent. These state changes are propagated via. event handlers. Often this state doesn't represent what's in the DOM or we just want to abstract the DOM from the consumer. We felt this format struct a good balance between readability and prediability.
+ */
+export type AbstractEvent<
```

Comment:
The `AbstractEvent` here is not an event, but a function that takes (handles) an event. I think `AbstractEventHandler` would make more sense.

Reply:
I like it!